addons/l10n_in_website_sale/controllers/main.py

```diff
@@ -0,0 +1,49 @@
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo.http import request
+from odoo import _
+
+
+class WebsiteSaleInherit(WebsiteSale):
+
+    def shop_address_submit(
+        self, partner_id=None, address_type='billing', use_same=None, callback=None,
+        required_fields=None, **form_data
+    ):
+        shop_address_submit = super().shop_address_submit(
+            partner_id, address_type, use_same, callback, required_fields, **form_data
+        )
+        partner_gst_treatment = form_data.get('o_gst_treatment')
+        order_sudo = request.website.sale_get_order()
+        partner_sudo, address_type = self._prepare_address_update(
+            order_sudo, partner_id=partner_id and int(partner_id), address_type=address_type
+        )
+        if partner_sudo.country_code == "IN" and request.env.company.country_code == "IN":
```

Comment:
```suggestion
        if request.env.company.account_fiscal_country_id.code == "IN":
```
Now we are giving user the selection then this should do the job. Cause we have overseas option too